algorithms/tree/fenwick_tree.py

```diff
@@ -0,0 +1,82 @@
+"""
+Fenwick Tree / Binary Indexed Tree
+
+Consider we have an array arr[0 . . . n-1]. We would like to
+1. Compute the sum of the first i elements.
+2. Modify the value of a specified element of the array arr[i] = x where 0 <= i <= n-1.
+
+A simple solution is to run a loop from 0 to i-1 and calculate the sum of the elements. To update a value, simply do arr[i] = x.
+The first operation takes O(n) time and the second operation takes O(1) time.
+Another simple solution is to create an extra array and store the sum of the first i-th elements at the i-th index in this new array.
+The sum of a given range can now be calculated in O(1) time, but the update operation takes O(n) time now.
+This works well if there are a large number of query operations but a very few number of update operations.
+
+
+There are two solutions that can perform both the query and update operations in O(logn) time.
+1. Fenwick Tree
+2. Segment Tree
+
+Compared with Segment Tree, Binary Indexed Tree requires less space and is easier to implement.
+"""
+
+def get_sum(bit_tree, i):
```

Comment:
Conver this into a class, to use it as a data structure.

Reply:
Updated in the new commit